erdos/ray/ray_cluster.py

```diff
@@ -0,0 +1,55 @@
+import json
+import re
+
+from erdos.cluster import Cluster
+
+
+class RayCluster(Cluster):
+    def __init__(self, redis_address="", can_install_ray=False):
+        super(RayCluster, self).__init__()
+        self.redis_address = redis_address
+        self.can_install_ray = can_install_ray
+
+    def __del__(self):
+        self.broadcast_command("ray stop")
+        super(RayCluster, self).__del__()
+
+    def install_ray(self, node):
+        node.run_command("pip install ray", timeout=180)
+
+    def ray_installed(self, node):
+        return b"not found" not in node.run_command("ray --help")
+
+    def setup_head_node(self, node):
+        resources = {node.name: 512}
+        ray_start_command = "ray start --head --resources '{}'".format(
+            json.dumps(resources))
+        response = node.run_command(ray_start_command).decode("utf-8")
+
+        matches = re.findall('redis_address="[0-9.:]+"', response)
+        if len(matches) != 1:
+            raise Exception("Unable to parse redis address")
+
+        self.redis_address = matches[0].split('"')[1]
+
+    def setup_worker(self, node):
+        resources = {node.name: 512}
```

Comment:
Do all the nodes have the same 512 of what? I'm not sure I understand this line.

Reply:
Ray can't assign more than 512 of a resource to a node.